iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Branch.java

```diff
@@ -57,16 +60,79 @@
 
     Side getSide(Terminal terminal);
 
-    CurrentLimits getCurrentLimits(Side side);
+    /**
+     * @deprecated Use {@link #getOperationalLimits1(LimitType, Class)} or {@link #getOperationalLimits2(LimitType, Class)} instead.
+     */
+    @Deprecated
+    default CurrentLimits getCurrentLimits(Side side) {
+        if (side == Side.ONE) {
+            return getOperationalLimits1(LimitType.CURRENT, CurrentLimits.class);
+        } else if (side == Side.TWO) {
+            return getOperationalLimits2(LimitType.CURRENT, CurrentLimits.class);
+        }
+        throw new AssertionError("Unexpected side: " + side);
+    }
 
-    CurrentLimits getCurrentLimits1();
+    /**
+     * @deprecated Use {@link #getOperationalLimits1(LimitType, Class)} instead.
+     */
+    @Deprecated
+    default CurrentLimits getCurrentLimits1() {
+        return getOperationalLimits1(LimitType.CURRENT, CurrentLimits.class);
+    }
+
+    default List<OperationalLimits> getOperationalLimits1() {
+        return Collections.singletonList(getCurrentLimits1());
+    }
+
+    default <L extends OperationalLimits> L getOperationalLimits1(LimitType limitType, Class<L> limitClazz) {
+        return limitType == LimitType.CURRENT && limitClazz == CurrentLimits.class ? (L) getCurrentLimits1() : null;
+    }
 
     CurrentLimitsAdder newCurrentLimits1();
 
-    CurrentLimits getCurrentLimits2();
+    default ApparentPowerLimitsAdder newApparentPowerLimits1() {
+        throw new UnsupportedOperationException();
+    }
+
+    default ActivePowerLimitsAdder newActivePowerLimits1() {
+        throw new UnsupportedOperationException();
+    }
+
+    default VoltageLimitsAdder newVoltageLimits1() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Are all these type of limits are possible for lines and/or transformers? I thought apparentPower was for transformers: we do not have this information for lines.

Reply:
apparentPower is voltage*current so it should be fine (ratedS is manufacturer information)